pkg/reconciler/managed/reconciler.go

```diff
@@ -631,6 +650,12 @@ func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (reconc
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 	}
 
+	if observation.ResourcePending {
```

Comment:
Is there a way to calculate this here instead of expecting controller to return? I wonder if we can say `ignore ResourceExists: false if the creation time annotation is set and X seconds delay has not passed yet` where X is given in `Setup` function of the controller. I guess the downside is that retries will have some delay (mostly 1-2s maybe?) but it'd let us not have to worry about eventually consistent APIs one by one when people open issues about leaking.